virttest/qemu_vm.py

```diff
@@ -1484,6 +1484,22 @@ def sort_key(dev):
         if params.get("defconfig", "yes") == "no":
             devices.insert(StrDev('nodefconfig', cmdline=" -nodefconfig"))
 
+        # Add device intel-iommu, it must be added before any virtio devices
+        if (params.get('intel_iommu') == 'yes' and
+                devices.has_device('intel-iommu')):
+            iommu_params = {
+                'intremap': params.get('iommu_intremap', 'on'),
+                'device-iotlb': params.get('iommu_device_iotlb', 'on'),
+                'caching-mode': params.get('iommu_caching_mode'),
+                'eim': params.get('iommu_eim'),
+                'x-buggy-eim': params.get('iommu_x_buggy_eim'),
+                'version': params.get('iommu_version'),
+                'x-scalable-mode': params.get('iommu_x_scalable_mode'),
+                'dma-drain': params.get('iommu_dma_drain'),
+                'pt': params.get('iommu_pt'),
+                'aw-bits': params.get('iommu_aw_bits')}
```

Comment:
In older qemu versions, some arguments are different. Please note to handle it if users want to test them in future.

$ /usr/libexec/qemu-kvm -version
QEMU emulator version 2.12.0 (qemu-kvm-rhev-2.12.0-33.el7_7.5)
Copyright (c) 2003-2017 Fabrice Bellard and the QEMU Project developers

$ /usr/libexec/qemu-kvm -device intel-iommu,?
intel-iommu.eim=OnOffAuto (on/off/auto)
**intel-iommu.x-aw-bits=uint8**
intel-iommu.x-buggy-eim=bool
intel-iommu.intremap=bool
intel-iommu.version=uint32
intel-iommu.pt=bool
intel-iommu.device-iotlb=bool
intel-iommu.caching-mode=bool

Reply:
@PaulYuuu Thanks, I guess it is less possible for us to test it.